lib/taskana-core/src/main/java/pro/taskana/TaskanaEngineConfiguration.java

```diff
@@ -489,31 +283,415 @@ public static boolean shouldUseLowerCaseForAccessIds() {
+      this.userRefreshJobRunEvery = userRefreshJobRunEvery;
+      return this;
+    }
+
+    public TaskanaEngineConfigurationBuilder userRefreshJobFirstRun(
+        Instant userRefreshJobFirstRun) {
+      this.userRefreshJobFirstRun = userRefreshJobFirstRun;
+      return this;
+    }
+
+    public TaskanaEngineConfigurationBuilder minimalPermissionsToAssignDomains(
+        List<WorkbasketPermission> minimalPermissionsToAssignDomains) {
+      this.minimalPermissionsToAssignDomains = minimalPermissionsToAssignDomains;
+      return this;
+    }
+
+    public TaskanaEngineConfiguration build() {
+      return new TaskanaEngineConfiguration(this);
+    }
+
+    public TaskanaEngineConfigurationBuilder initTaskanaProperties() {
+      return this.initTaskanaProperties(TASKANA_PROPERTIES, TASKANA_PROPERTY_SEPARATOR);
+    }
+
+    public TaskanaEngineConfigurationBuilder initTaskanaProperties(String propertiesFile) {
+      return this.initTaskanaProperties(propertiesFile, TASKANA_PROPERTY_SEPARATOR);
+    }
+
+    public TaskanaEngineConfigurationBuilder initTaskanaProperties(
+        String propertiesFile, String separator) {
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug(
+            "Reading taskana configuration from {} with separator {}", propertiesFile, separator);
+      }
+      loadProperties(propertiesFile);
+      configureAnnotatedFields(this, separator, properties);
+      roleMap = configureRoles(separator, properties, shouldUseLowerCaseForAccessIds());
+      classificationCategoriesByTypeMap =
+          configureClassificationCategoriesForType(properties, classificationTypes);
+
+      if (LOGGER.isDebugEnabled()) {
+        roleMap.forEach((k, v) -> LOGGER.debug("Found Taskana RoleConfig {} : {} ", k, v));
+        LOGGER.debug(
+            "Configured number of task and workbasket updates per transaction: {}", jobBatchSize);
+        LOGGER.debug("Number of retries of failed task updates: {}", maxNumberOfJobRetries);
+        LOGGER.debug("CleanupJob configuration: first run at {}", cleanupJobFirstRun);
+        LOGGER.debug("CleanupJob configuration: runs every {}", cleanupJobRunEvery);
+        LOGGER.debug(
+            "CleanupJob configuration: minimum age of tasks to be cleanup up is {}",
+            cleanupJobMinimumAge);
+        LOGGER.debug(
+            "TaskCleanupJob configuration: all completed task with the "
+                + "same parent business property id {}",
+            taskCleanupJobAllCompletedSameParentBusiness);
+        LOGGER.debug(
+            "Configured classification categories : {}", classificationCategoriesByTypeMap);
+        LOGGER.debug("Configured domains: {}", domains);
+        LOGGER.debug("Configured classificationTypes: {}", classificationTypes);
+        LOGGER.debug("Configured custom Holidays : {}", customHolidays);
+        LOGGER.debug(
+            "Configured minimalPermissionsToAssignDomains : {}", minimalPermissionsToAssignDomains);
+      }
```

Comment:
this is probably incomplete. 
Idea: Let's create a debug statement during the TaskanaEngine initalization, so that the configuration can be viewed during startup.